packages/graphql-auth-transformer/src/resources.ts

```diff
@@ -689,4 +775,117 @@ identityField: "${rule.identityField || DEFAULT_IDENTITY_FIELD}" }`
     public setUserGroups(): SetNode {
         return set(ref('userGroups'), raw('$util.defaultIfNull($ctx.identity.claims.get("cognito:groups"), [])'));
     }
+
+    public getAuthModeCheckWrappedExpression(expectedAuthModes: Set<AuthProvider>, expression: Expression): Expression {
+        if (!expectedAuthModes || expectedAuthModes.size === 0) {
+            return expression;
+        }
+
+        const conditions = [];
+
+        for (const expectedAuthMode of expectedAuthModes) {
+            conditions.push(equals(
+                ref(ResourceConstants.SNIPPETS.AuthMode),
+                str(`${expectedAuthMode}`)
+            ));
+        }
+
+        return block("Check authMode and execute owner/group checks", [
+            iff(
+                conditions.length === 1 ? conditions[0] : or (conditions),
+                expression
+            )
+        ]);
+    }
+
+    public getAuthModeDeterminationExpression(authProviders: Set<AuthProvider>): Expression {
+        if (!authProviders || authProviders.size === 0) {
+            return comment(`No authentication mode determination needed`);
+        }
+
+        const expressions = [];
+
+        for (const authProvider of authProviders) {
+
+            if (authProvider === 'userPools') {
+                const userPoolsExpression = iff(
```

Comment:
This looks unnecessarily complicated to me. At the resolver layer do we need to do anything different when handling OIDC & Cognito? Both of  those auth mechanisms will have similar things and the current @auth owner & group rules use values in `identity.claims()` which is included in both auth modes. In other words, I don't think you need to do anything different at the resolver layer to authorize OIDC vs User Pools. When I use `{ allow: owner, identityField: "sub", provider: OIDC }`, the resolver logic is no different than if I did `{ allow: owner, identityField: "sub", provider: userPools }`. If you needed to you could probably just look for the existence of "sub" & "issuer" to determine that there was a valid JWT provided was provided.

Reply:
For this check I exactly checking what the AppSync service is doing to determine the authentication type. By raising this question I was wondering if the existing code supports this:
```
{ allow owner, identityField: "sub", provider: oidc, ownerField: "oidcOwner"  }
{ allow owner, identityField: "sub", provider: userPools}
```
Because it became a possible scenario and it do requires to differentiate between ```userPools``` and ```oidc``` since unique ```sub```s are not guaranteed between different identity providers.